apps/typegpu-docs/src/content/examples/simulation/gravity/render.ts

```diff
@@ -0,0 +1,125 @@
+import tgpu from 'typegpu';
+import * as d from 'typegpu/data';
+import * as std from 'typegpu/std';
+import { radiusOf } from './helpers.ts';
+import {
+  renderBindGroupLayout as renderLayout,
+  renderSkyBoxBindGroupLayout as skyBoxLayout,
+  VertexOutput,
+} from './schemas.ts';
+
+export const skyBoxVertex = tgpu['~unstable']
+  .vertexFn({
+    in: {
+      position: d.vec4f,
+      uv: d.vec2f,
+    },
+    out: {
+      pos: d.builtin.position,
+      texCoord: d.vec3f,
+    },
+  })((input) => {
+    const viewRotationMatrix = d.mat4x4f(
+      skyBoxLayout.$.camera.view.columns[0],
+      skyBoxLayout.$.camera.view.columns[1],
+      skyBoxLayout.$.camera.view.columns[2],
+      d.vec4f(0, 0, 0, 1),
+    );
+    return {
+      pos: std.mul(
+        skyBoxLayout.$.camera.projection,
+        std.mul(viewRotationMatrix, input.position),
+      ),
+      texCoord: input.position.xyz,
+    };
```

Comment:
Instead of creating a new matrix, we can "turn off" translation by setting the 4th parameter of the position vector to `0`.

```suggestion
    return {
      pos: std.mul(
        skyBoxLayout.$.camera.projection,
        std.mul(
          skyBoxLayout.$.camera.view,
          d.vec4f(input.position.xyz, 0),
        ),
      ),
      texCoord: input.position.xyz,
    };
```

Reply:
From what I see, you also need to add (0, 0, 0, w) to the intermediate result, like this: 
```typescript
return {
      pos: std.mul(
        skyBoxLayout.$.camera.projection,
        std.add(
          std.mul(
            skyBoxLayout.$.camera.view,
            d.vec4f(input.position.xyz, 0),
          ),
          d.vec4f(0, 0, 0, input.position.w),
        ),
      ),
      texCoord: input.position.xyz,
    };
```
or
```typescript
    return {
      pos: std.mul(
        skyBoxLayout.$.camera.projection,
        std.add(
          std.mul(
            skyBoxLayout.$.camera.view,
            d.vec4f(std.div(input.position.xyz, input.position.w), 0),
          ),
          d.vec4f(0, 0, 0, 1),
        ),
      ),
      texCoord: input.position.xyz,
    };
```

Without this, the skybox doesn't seem to work.